src/trx-generator.ts

```diff
@@ -100,16 +123,92 @@ const renderTestSuiteResult = (
   testEntriesNode: XMLElementOrXMLNode,
   resultsNode: XMLElementOrXMLNode,
   computerName: string,
+  postProcessTestResult?: [
+    (
+      testSuiteResult: JestTestSuiteResult,
+      testResult: JestTestResult,
+      testResultNode: XMLElementOrXMLNode,
+    ) => void
+  ],
 ) => {
   const perTestDuration = getSuitePerTestDuration(testSuiteResult);
   const perTestDurationFormatted = formatDuration(perTestDuration);
 
-  testSuiteResult.testResults.forEach((testResult, index) => {
+  if (testSuiteResult.testResults && testSuiteResult.testResults.length) {
+    testSuiteResult.testResults.forEach((testResult, index) => {
+      const testId = uuid.v4();
+      const executionId = uuid.v4();
+      const fullTestName = getFullTestName(testResult);
+
+      // UnitTest
+      const unitTest = testDefinitionsNode
+        .ele("UnitTest")
+        .att("name", fullTestName)
+        .att("id", testId);
+      unitTest.ele("Execution").att("id", executionId);
+      unitTest
+        .ele("TestMethod")
+        .att("codeBase", `Jest_${fullTestName}`)
+        .att("name", fullTestName)
+        .att("className", getTestClassName(testResult));
+
+      // TestEntry
+      testEntriesNode
+        .ele("TestEntry")
+        .att("testId", testId)
+        .att("executionId", executionId)
+        .att("testListId", testListNotInListId);
+
+      // UnitTestResult
+      const result = resultsNode
+        .ele("UnitTestResult")
+        .att("testId", testId)
+        .att("executionId", executionId)
+        .att("testName", fullTestName)
+        .att("computerName", computerName)
+        .att("duration", perTestDurationFormatted)
+        .att(
+          "startTime",
+          new Date(
+            testSuiteResult.perfStats.start + index * perTestDuration,
+          ).toISOString(),
+        )
+        .att(
+          "endTime",
+          new Date(
+            testSuiteResult.perfStats.start + (index + 1) * perTestDuration,
+          ).toISOString(),
+        )
+        .att("testType", testType)
+        .att("outcome", testOutcomeTable[testResult.status])
+        .att("testListId", testListNotInListId);
+
+      if (testResult.status === "failed") {
+        result
+          .ele("Output")
+          .ele("ErrorInfo")
+          .ele(
+            "Message",
+            sanitizeString(testResult.failureMessages.join("\n")),
+          );
+      }
+
+      // Perform any post processing for this test result.
+      if (postProcessTestResult) {
+        postProcessTestResult.forEach(postProcess =>
+          postProcess(testSuiteResult, testResult, result),
+        );
+      }
+    });
+  } else if (testSuiteResult.failureMessage) {
+    // For suites that failed to run, we will generate a test result that documents the failure.
+    // This occurs when their is a failure compiling/loading the suite, not when a test in the suite fails.
```

Comment:
Typo: there

Reply:
Sure I will get it updated and add some tests. Thanks for the project, it was very easy to integrate.